stock_picking_mass_action/wizard/mass_action.py

```diff
@@ -0,0 +1,128 @@
+        help="""check this box if you want to check the availability of"""
+        """ the selected Pickings.""")
+
+    force_availability = fields.Boolean(
+        string='Force Availability', default=_default_force_availability,
+        help="""check this box if you want to force the availability"""
+        """ of the selected Pickings.""")
+
+    transfer = fields.Boolean(
+        string='Transfer', default=_default_transfer,
+        help="""check this box if you want to transfer all the selected"""
+        """ pickings.\n You'll not have the possibility to realize a"""
+        """ partial transfer.\n If you want  to do that, please do it"""
+        """ manually on the picking form.""")
+
+    create_invoice = fields.Boolean(
+        'Create Invoices/Refunds', default=_default_create_invoice,
+        help="""check this box if you want to create Invoices or Refunds for"""
+        """ all the selected Pickings.""")
+
+    @api.multi
+    def mass_action(self):
+        self.ensure_one()
+        picking_obj = self.env['stock.picking']
+        transfert_wizard_obj = self.env['stock.transfer_details']
+        picking_ids = self.env.context.get('active_ids')
+
+        # Get draft pickings and confirm them if asked
+        if self.confirm:
+            domain = [('state', '=', 'draft'),
+                      ('id', 'in', picking_ids)]
+            draft_picking_lst = picking_obj.search(domain, order='min_date')
+            draft_picking_lst.action_confirm()
+
+        # Get confirmed pickings
+        domain = [('state', 'in', ['confirmed', 'partially_available']),
+                  ('id', 'in', picking_ids)]
+        confirmed_picking_lst = picking_obj.search(domain, order='min_date')
+
+        # check availability if asked
+        if self.check_availability:
+            confirmed_picking_lst.check_assign_all()
+
+        # Force availability if asked
+        if self.force_availability:
+            confirmed_picking_lst.force_assign()
+
+        # Get all pickings ready to transfer and transfer them if asked
+        if self.transfer:
+            domain = [('state', '=', 'assigned'),
+                      ('id', 'in', picking_ids)]
+            assigned_picking_lst = picking_obj.search(domain, order='min_date')
+            for picking in assigned_picking_lst:
+                transfert_wizard = transfert_wizard_obj.create(
+                    {'picking_id': picking.id})
+                transfert_wizard.do_detailed_transfer()
+
+        # Get all pickings ready to invoice and invoice them if asked
+        if self.create_invoice:
+            domain = [('invoice_state', '=', '2binvoiced'),
+                      ('id', 'in', picking_ids)]
+            to_invoice_picking_lst = picking_obj.search(
+                domain, order='min_date')
+            ctx = self.env.context.copy()
+            ctx['active_ids'] = [x.id for x in to_invoice_picking_lst]
```

Comment:
`to_invoice_picking_lst.ids` is simpler
(I don't think it is necessary to add a `_lst` suffix, we can use `to_invoice_pickings` which speaks by itself)